diff --git a/net/ipv4/tcp_bbr.c b/net/ipv4/tcp_bbr.c
index a180fa648..8ab0ad449 100644
--- a/net/ipv4/tcp_bbr.c
+++ b/net/ipv4/tcp_bbr.c
@@ -212,9 +212,9 @@ static const int bbr_pacing_margin_percent = 1;
  * and send the same number of packets per RTT that an un-paced, slow-starting
  * Reno or CUBIC flow would:
  */
-static const int bbr_startup_pacing_gain = BBR_UNIT * 277 / 100 + 1;
+static const int bbr_startup_pacing_gain = BBR_UNIT * 2885 / 1000 + 1;
 /* The gain for deriving startup cwnd: */
-static const int bbr_startup_cwnd_gain = BBR_UNIT * 2;
+static const int bbr_startup_cwnd_gain = BBR_UNIT * 2885 / 1000 + 1;
 /* The pacing gain in BBR_DRAIN is calculated to typically drain
  * the queue created in BBR_STARTUP in a single round:
  */
@@ -269,7 +269,7 @@ static const u32 bbr_ecn_max_rtt_us = 5000;
 /* On losses, scale down inflight and pacing rate by beta scaled by BBR_SCALE.
  * No loss response when 0.
  */
-static const u32 bbr_beta = BBR_UNIT * 30 / 100;
+static const u32 bbr_beta;
 
 /* Gain factor for ECN mark ratio samples, scaled by BBR_SCALE (1/16 = 6.25%) */
 static const u32 bbr_ecn_alpha_gain = BBR_UNIT * 1 / 16;
@@ -282,12 +282,12 @@ static const u32 bbr_ecn_alpha_init = BBR_UNIT;
 /* On ECN, cut inflight_lo to (1 - ecn_factor * ecn_alpha) scaled by BBR_SCALE.
  * No ECN based bounding when 0.
  */
-static const u32 bbr_ecn_factor = BBR_UNIT * 1 / 3;	 /* 1/3 = 33% */
+static const u32 bbr_ecn_factor;
 
 /* Estimate bw probing has gone too far if CE ratio exceeds this threshold.
  * Scaled by BBR_SCALE. Disabled when 0.
  */
-static const u32 bbr_ecn_thresh = BBR_UNIT * 1 / 2;  /* 1/2 = 50% */
+static const u32 bbr_ecn_thresh;
 
 /* If non-zero, if in a cycle with no losses but some ECN marks, after ECN
  * clears then make the first round's increment to inflight_hi the following
@@ -296,24 +296,24 @@ static const u32 bbr_ecn_thresh = BBR_UNIT * 1 / 2;  /* 1/2 = 50% */
 static const u32 bbr_ecn_reprobe_gain = BBR_UNIT * 1 / 2;
 
 /* Estimate bw probing has gone too far if loss rate exceeds this level. */
-static const u32 bbr_loss_thresh = BBR_UNIT * 2 / 100;  /* 2% loss */
+static const u32 bbr_loss_thresh = BBR_UNIT;
 
 /* Slow down for a packet loss recovered by TLP? */
-static const bool bbr_loss_probe_recovery = true;
+static const bool bbr_loss_probe_recovery;
 
 /* Exit STARTUP if number of loss marking events in a Recovery round is >= N,
  * and loss rate is higher than bbr_loss_thresh.
  * Disabled if 0.
  */
-static const u32 bbr_full_loss_cnt = 6;
+static const u32 bbr_full_loss_cnt;
 
 /* Exit STARTUP if number of round trips with ECN mark rate above ecn_thresh
  * meets this count.
  */
-static const u32 bbr_full_ecn_cnt = 2;
+static const u32 bbr_full_ecn_cnt;
 
 /* Fraction of unutilized headroom to try to leave in path upon high loss. */
-static const u32 bbr_inflight_headroom = BBR_UNIT * 15 / 100;
+static const u32 bbr_inflight_headroom = BBR_UNIT * 5 / 100;
 
 /* How much do we increase cwnd_gain when probing for bandwidth in
  * BBR_BW_PROBE_UP? This specifies the increment in units of
@@ -329,7 +329,7 @@ static const u32 bbr_bw_probe_cwnd_gain = 1;
  * We aim to be fair with Reno/CUBIC up to a BDP of at least:
  *  BDP = 25Mbps * .030sec /(1514bytes) = 61.9 packets
  */
-static const u32 bbr_bw_probe_max_rounds = 63;
+static const u32 bbr_bw_probe_max_rounds = 8;
 
 /* Max amount of randomness to inject in round counting for Reno-coexistence.
  */
@@ -339,10 +339,10 @@ static const u32 bbr_bw_probe_rand_rounds = 2;
  * We aim to be fair with Reno/CUBIC up to an inter-loss time epoch of at least:
  *  BDP*RTT = 25Mbps * .030sec /(1514bytes) * 0.030sec = 1.9 secs
  */
-static const u32 bbr_bw_probe_base_us = 2 * USEC_PER_SEC;  /* 2 secs */
+static const u32 bbr_bw_probe_base_us = USEC_PER_SEC / 2;
 
 /* Use BBR-native probes spread over this many usec: */
-static const u32 bbr_bw_probe_rand_us = 1 * USEC_PER_SEC;  /* 1 secs */
+static const u32 bbr_bw_probe_rand_us = USEC_PER_SEC / 4;
 
 /* Use fast path if app-limited, no loss/ECN, and target cwnd was reached? */
 static const bool bbr_fast_path = true;
@@ -1388,7 +1388,7 @@ static void bbr_adapt_lower_bounds(struct sock *sk,
 	}
 
 	/* Loss response. */
-	if (bbr->loss_in_round) {
+	if (bbr->loss_in_round && bbr_param(sk, beta)) {
 		bbr_init_lower_bounds(sk, true);
 		bbr_loss_lower_bounds(sk, &bbr->bw_lo, &bbr->inflight_lo);
 	}
